### PR TITLE
feat: new warning UI in toolbar

### PR DESCRIPTION
**Pull request by just-toby**
adds a warning UI to the expander on the right when the toolbar is collapsed  - IF there is a price impact warning to view in the expanded state.

Screenshot 2023-03-20 at 11 14 22 AM

Screenshot 2023-03-20 at 11 14 16 AM

### CODE REVIEW

File: src/components/Swap/Toolbar/Caption.tsx [hunk 1]
```diff
@@ -151,59 +174,38 @@ export function Trade({
   gasUseEstimateUSD,
   expanded,
   loading,
+  warning,
 }: TradeProps & TradeTooltip & ExpandProps) {
+  const widgetWidth = useWidgetWidth()
+  const shouldHideUSD = widgetWidth < 360 && warning && !expanded
```
Comment (just-toby) [inline src/components/Swap/Toolbar/Caption.tsx:180]: this behavior is based on a conversation with Fred  - at narrow widget width, we need to shorten the Trade Price string length so the new button fits

File: src/components/Swap/PriceImpactRow.tsx [hunk 2]
```diff
@@ -19,7 +19,9 @@ export function PriceImpactRow({ impact, reverse, tooltipText }: PriceImpactProp
   return (
     <Row gap={0.25} flex align="center" flow={reverse ? 'row-reverse' : 'row wrap'}>
       <ThemedText.Body2 userSelect={false} color={impact.warning ?? 'hint'}>
-        <TooltipText text={`(${formatPriceImpact(impact?.percent)})`}>{tooltipText}</TooltipText>
+        <TooltipText text={`(${formatPriceImpact(impact?.percent)})`}>
+          <ThemedText.Caption>{tooltipText}</ThemedText.Caption>
```
Comment (zzmp) [inline src/components/Swap/PriceImpactRow.tsx:23]: Is this change related to the warning UI? Is it necessary?

File: src/components/Swap/Toolbar/Caption.tsx [hunk 3]
```diff
@@ -139,10 +137,35 @@ export interface TradeProps {
 
 interface ExpandProps {
   expanded: boolean
-}
+  warning?: 'warning' | 'error'
+}
+
+const ExpanderRow = styled(Row)<{ $expanded: boolean; warning?: 'warning' | 'error' }>`
+  ${({ warning, $expanded }) =>
+    warning &&
```
Comment (zzmp) [inline src/components/Swap/Toolbar/Caption.tsx:145]: This is hard to follow because it's using defaulting - could you use an if-statement instead?

```
  ${({ warning, $expanded }) => {
    if (!warning) return undefined
    return css`
```

or

```
  if (warning) {
    return css`
      ...
    `
  }

File: src/components/Swap/Toolbar/Caption.tsx [hunk 4]
```diff
@@ -139,10 +137,35 @@ export interface TradeProps {
...
+`
 
-function Expander({ expanded }: ExpandProps) {
-  return <ExpandIcon $expanded={expanded} />
+function Expander({ expanded, warning }: ExpandProps) {
+  return (
+    <ExpanderRow $expanded={expanded} warning={warning} gap={0.5}>
+      {warning && !expanded && (
+        <Tooltip icon={AlertTriangle} iconProps={{ color: warning }} placement="auto">
+          <ThemedText.Caption>
```
Comment (zzmp) [inline src/components/Swap/Toolbar/Caption.tsx:161]: Is this needed, or is it already supplied by `Tooltip`?

File: src/components/Swap/Toolbar/Caption.tsx [hunk 5]
```diff
@@ -139,10 +137,35 @@ export interface TradeProps {
...
+            <Trans>Your trade will have a high impact on the market price of this pool.</Trans>
+          </ThemedText.Caption>
+        </Tooltip>
+      )}
+      <ExpandIcon $expanded={expanded} color={expanded ? undefined : warning} />
+    </ExpanderRow>
+  )
 }
 
 export function Trade({
```
Comment (zzmp) [inline src/components/Swap/Toolbar/Caption.tsx:171]: Modifications to Trade should go in a separate PR, I think. Are they related?

File: src/components/Swap/Toolbar/Caption.tsx (hunk 1 above)
Comment (zzmp) [reply src/components/Swap/Toolbar/Caption.tsx:180]: Is there a constant that can be used instead of `360`?
You should include a comment above this line explaining this behavior. A good rule of thumb is that if you leave a PR comment, it should probably be noted in the code.

**Review by zzmp — CHANGES_REQUESTED** [review]

File: src/components/Swap/Toolbar/Caption.tsx (hunk 4 above)
Comment (just-toby) [reply src/components/Swap/Toolbar/Caption.tsx:161]: this is needed

File: src/components/Swap/Toolbar/Caption.tsx (hunk 1 above)
Comment (just-toby) [reply src/components/Swap/Toolbar/Caption.tsx:180]: updated to use a share constant that already exists

File: src/components/Swap/PriceImpactRow.tsx (hunk 2 above)
Comment (just-toby) [reply src/components/Swap/PriceImpactRow.tsx:23]: it's semi-related because all the tooltips should have consistent font size and the warning UI includes tooltips

File: src/components/Swap/Toolbar/Caption.tsx [hunk 6]
```diff
@@ -151,59 +176,38 @@ export function Trade({
   gasUseEstimateUSD,
   expanded,
   loading,
+  warning,
 }: TradeProps & TradeTooltip & ExpandProps) {
+  const widgetWidth = useWidgetWidth()
+  const shouldHideUSD = widgetWidth < WIDGET_BREAKPOINTS.EXTRA_SMALL && warning && !expanded
```
Comment (zzmp) [inline src/components/Swap/Toolbar/Caption.tsx:182]: Add a comment above this line explaining why it is necessary to hide USD.

**Review by zzmp — APPROVED** [review]
